lib/network_interface/worker.ex

```diff
@@ -13,45 +13,105 @@
 # limitations under the License.
 
 defmodule Nerves.NetworkInterface.Worker do
+  @moduledoc """
+  Working for NetworkInterface. 
+  See `Nerves.NetworkInterface` for more details.
+  """
+
   use GenServer
   require Logger
 
-  @moduledoc false
+  defstruct port: nil
+
+  @typedoc "State of the GenServer"
+  @type state :: %__MODULE__{port: port}
```

Comment:
> It seems to be idiomatic to call a struct type t. I understand the use of state as attempting to give this an intention revealing name, but I thought it would be worth considering.

This is personal to me. If it helps more people we can change it. Again purely subjective and maybe something that just stems from my OOP days.
when i have a module that is purely data, i will call it `t` and if i have something statefull i will usually call it `state`. (ive been known to break even my own convention.) :+1:  on changing it to `t`